### DISCUSSION TITLE
BirdNet-Pi /portable

### DISCUSSION THREAD

**Discussion by stevedee** [mcguirepr89/BirdNET-Pi#1075] [Show and tell]
I set up a system for portable operation yesterday and did a quick 30 minute test at our local RSPB reserve.

![BirdNetP-portable](https://github.com/mcguirepr89/BirdNET-Pi/assets/1870807/b2226e90-df26-453f-be2c-5bbac3deca63)

As expected, while walking along with the unit poking out through the netting of my rucksack, I probably generated too much noise for any sensible detecting. But while I just stood looking around through my binoculars, the system saved approximately 100 recordings, mostly wigeon, curlew, green-winged teal and a few wren.

The only system modification was to add my hot-spot details to the _wpa_supplicant.conf_ file. The system auto switches between home & portable operation based upon best wiFi signal, and http://birdnetpi-2.local provides web access on my phone in both cases.

**Comment by Rimbao** [comment]
Hi stevedee, please explain your  configuration - switches between home & portable operation based upon best wiFi signal.

Look there https://github.com/mcguirepr89/BirdNET-Pi/discussions/39#discussioncomment-7262643

Many Thanks Rimbao

**Comment by stevedee** [comment]
Hi Rimbao,
Just to be clear, I'm not bothered about WiFi switching based upon the best WiFi signal.

When I'm at home, the phone hot-spot is disabled, and both my phone and my portable BirdNet system use home WiFi.
When I reach a remote location, I enable the hot-spot on my phone and power-up the BirdNet Pi.
So BirdNetPi picks the appropriate WiFi based upon availability.

In order to create the hot-spot information for the Pi file **_/etc/wpa_supplicant/wpa_supplicant.conf_** I ran this command on a Linux computer terminal:-
**sudo wpa_passphrase {yourSSID} {yourPassword}**

For example:-

**sudo wpa_passphrase "HOTSPOT PENGUIN" "ImGoingMobile"**

...would produce and output like this:-

**network={
	ssid="HOTSPOT PENGUIN"
	#psk="ImGoingMobile"
	psk=906ddcc1763ad15234061fb92cccf4d606410e0566cdfe4ac5695c15dd3dd78f
}**



So the modified wpa_supplicant.conf may look like this:-

ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
update_config=1

network={
	ssid="HOTSPOT PENGUIN"
	#psk="ImGoingMobile"
	psk=906ddcc1763ad15234061fb92cccf4d606410e0566cdfe4ac5695c15dd3dd78f
}

network={
	ssid="BIG PENGUIN"
	#psk="ticklemepink"
	psk=15575f79ed2fe6f51308d216183a5d265f2949780d4a35e1b520674c531cd78f
}

...where BIG PENGUIN would be the home WiFi

I hope this answers your question, but please come back if I'm missing your point.
Regards,
SteveDee

**Comment by ClemensGruber** [comment]
Many thanks for the hint @stevedee, that a mobile phone hotspot is also a local WiFi and so monitoring of the Pi is so also possible this way. Without configurating the RasPi expicitly as an AP! On my side (German mobile phone network and a up to date Sony phone with Android) access via http://birdnetpi.local was not possible, I had to start the BirdNetPi first on a monitor and keyboard and had to figure out the local IP via `ifconfig`. After that it was possible to access the BirdNetPi on my mobile phone but only with the IP address! Don't know if the phone (up to date Android) or a RasPi configuration is the problem.

**Reply by MueJosh** [reply]
Maybe, just maybe: 

I have a kinda similar issue (it's not an issue, I just don't know the right wording) when I use vpn to ssh into my pi 
I can't use birndetpi.local only the ip address OR birdnetpi ( http://birdnetpi ) 

Does the second option work for you?

**Reply by stevedee** [reply]
Name resolution (i.e. http://birdnet.local) won't work if using phone as an access point.

But what I do is use an Android phone app called **_Network Tools Library_** to tell me the Pi ip address, then type this into browser.

For more details, see: http://captainbodgit.blogspot.com/2023/11/birdnet-pi-portable-operation-mics-bird.html

**Comment by ClemensGruber** [comment]
I tried both http://birdnetpi and birdnetpi without protocol but same issue. I can not find a tool named "Network Tools Library" in my (German localized?) Android store, so perhaps it is not available or not for my device. But good to know that bare IP is working and hostname not.

**Reply by stevedee** [reply]
Sorry ClemensGruber, I forgot to mention that this app is not on Google Play. You will have to install **F-Droid** and then down-load it from there.

Go to: https://f-droid.org/ to install F-Droid
Then search F-Droid for: **_Network Tools Library_** and install from there.

You may find many useful Apps on F-Droid.

_Note: my phone runs on LineageOS (a de-Googled version of Android) so I never use Google Play apps_

**Reply by ClemensGruber** [reply]
I'm using now Network Analyzer, https://play.google.com/store/apps/details?id=net.techet.netanalyzerlite.an (the linked light version without any fee is sufficient) and in the app LAN scan to get the IP address of the RasPi.
